lib/ansible/plugins/action/fetch.py

```diff
@@ -161,29 +170,63 @@ def run(self, tmp=None, task_vars=None):
 
             dest = os.path.normpath(dest)
 
+            re_encrypt_needed = False
+            local_encrypted = False
+            b_file_path = to_bytes(dest, errors='surrogate_or_strict')
+            if self._loader.path_exists(b_file_path) and self._loader.is_file(b_file_path):
+                with open(to_bytes(dest), 'rb') as f:
+                    if self._loader._vault.is_encrypted(f.read(len(b_HEADER))):
+                        f.seek(0)
+                        data = f.read()
+                        if not self._loader._vault.secrets:
+                            raise AnsibleParserError("A vault password or secret must be specified to decrypt %s" % to_native(dest))
```

Comment:
`match_encrypt_secret` gives an error, so I think this extra handling can be deleted (or if not, this should probably return a failed result or raise an AnsibleActionFail instead of raising an AnsibleParserError).

```
fatal: [localhost]: FAILED! => {
    "msg": "Did not find a match for --encrypt-vault-id=default in the known vault-ids []"
}
```

Reply:
You are right, this is redundant.

Reply:
Fixed.